src/backend/commands/explain.c

```diff
@@ -1467,81 +1424,77 @@ ExplainNode(PlanState *planstate, List *ancestors,
 		case T_Motion:
 			{
 				Motion		*pMotion = (Motion *) plan;
+				SliceTable *sliceTable = planstate->state->es_sliceTable;
+
+				/* Descending into a new slice. */
+				if (sliceTable)
+				{
+					es->currentSlice = (Slice *) list_nth(sliceTable->slices,
+														  pMotion->motionID);
+				}
 
 				Assert(plan->lefttree);
 				Assert(plan->lefttree->flow);
 
+				/* Size of the child slice's gang */
 				motion_snd = es->currentSlice->gangSize;
-				motion_recv = (parentSlice == NULL ? 1 : parentSlice->gangSize);
 
-				/* scale the number of rows by the number of segments sending data */
-				scaleFactor = motion_snd;
+				/*
+				 * Size of the current slice's gang
+				 * (save_currentSlice->gangSize)
+				 */
+				motion_recv = scaleFactor;
 
 				switch (pMotion->motionType)
 				{
 					case MOTIONTYPE_HASH:
 						sname = "Redistribute Motion";
+
+						/*
+						 * scale the number of rows by the number of segments
+						 * sending data
+						 */
+						scaleFactor = motion_snd;
 						break;
 					case MOTIONTYPE_FIXED:
 						if (pMotion->isBroadcast)
 						{
 							sname = "Broadcast Motion";
+
+							/*
+							 * Scale the number of rows by the number of
+							 * segments receiving data, because they were
+							 * multiplied by this number at the planning
```

Comment:
> because they were multiplied by this number at the planning stage

Can you remember this codepart in the Postgres planner? This comment sounds weird.

Reply:
Seems like an outdated comment, scale_factor isn't changed in this block, I'll update this comment

Reply:
I updated the comment and now it says that the scaling was removed bc orca count rows number correctly

Reply:
What about Postgres planner? It's common code part.